pkg/operator/ceph/cluster/crash/crash.go

```diff
@@ -123,7 +124,60 @@ func (r *ReconcileNode) createOrUpdateCephCrash(node corev1.Node, tolerations []
 }
 
 // createOrUpdateCephCron is a wrapper around controllerutil.CreateOrUpdate
-func (r *ReconcileNode) createOrUpdateCephCron(cephCluster cephv1.CephCluster, cephVersion *version.CephVersion) (controllerutil.OperationResult, error) {
+func (r *ReconcileNode) createOrUpdateCephCronV1(cephCluster cephv1.CephCluster, cephVersion *version.CephVersion) (controllerutil.OperationResult, error) {
```

Comment:
A lot of repetitive code in `createOrUpdateCephCronV1` and `createOrUpdateCephCronBeta1`. Also in  `reconcileCrashRetentionCronV1` and `reconcileCrashRetentionCronV1Beta1`.

Reply:
I agree we have lot's of duplicate code but supporting both v1 and v1beta will have some duplicate codes.
looking` One suggestion would be to inject cronJob as a dependency in the method parameters.`

Reply:
I think we can make it work without code duplication using type assertion.

Consider the following code block as a reference:

```go
func (r *ReconcileNode) createOrUpdateCephCron(cephCluster cephv1.CephCluster, cephVersion *version.CephVersion) (controllerutil.OperationResult, error) {

	var cronJobInterface interface{}
	cronJobInterface = &v1beta1.CronJob{
		ObjectMeta: metav1.ObjectMeta{
			Name:      prunerName,
			Namespace: cephCluster.GetNamespace(),
		},
	}

	if kubeVersion >= 1.15 { {
		cronJobInterface = &v1.CronJob{
			ObjectMeta: metav1.ObjectMeta{
				Name:      prunerName,
				Namespace: cephCluster.GetNamespace(),
			},
		}
	}

	// Reflect type
	cronJob, ok := cronJobInterface.(*v1beta1.CronJob)
	if !ok {
		cronJob, ok = cronJobInterface.(*v1beta1.CronJob)
		if !ok {
			panic("unknown type!")
		}
	}
```